### PR TITLE
mgr/alerts: interval can be None

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Fixes: https://tracker.ceph.com/issues/48972

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/alerts/module.py [hunk 1]
```diff
@@ -180,8 +180,8 @@ def serve(self):
                     self._send_alert(new_status, diff)
                 last_status = new_status
 
-            self.log.debug('Sleeping for %d seconds', self.interval)
-            ret = self.event.wait(self.interval)
+            self.log.debug('Sleeping for %s seconds', self.interval)
```
Comment (tchaikov) [inline src/pybind/mgr/alerts/module.py:194]: `self.interval` should be a float actually, is 

> mgr/alerts: "MGR_MODULE_ERROR: Module 'alerts' has failed: %d format: a number is required, not str"

reproducible ? if yes, how?

Comment (sebastian-philipp) [reply src/pybind/mgr/alerts/module.py:194]: I have not reproduced this,  no.

Comment (sebastian-philipp) [reply src/pybind/mgr/alerts/module.py:194]: https://rook-io.slack.com/archives/C46Q5UC05/p1590494007085100

Comment (tchaikov) [reply src/pybind/mgr/alerts/module.py:194]: seems i need a `@rook.io` mail address for joining this workspace. =(

Comment (sebastian-philipp) [reply src/pybind/mgr/alerts/module.py:194]: in any case, imo `%s` is the safest variant, as it supports strings, floats and ints. Makes little sense to force a type in the debug statement.

**Comment by sebastian-philipp** [comment]
@tchaikov what do you think? ok for you?

File: src/pybind/mgr/alerts/module.py (hunk 1 above)
Comment (tchaikov) [reply src/pybind/mgr/alerts/module.py:194]: agreed. but i don't want to cover a bug as settings are supposed to be converted to the type as advertised by the "type" fields in the corresponding OPTIONS element.

**Review by tchaikov — APPROVED** [review]

File: src/pybind/mgr/alerts/module.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/alerts/module.py:194]: agreed. My idea is that this PR creates a better error message. At lease prints a debug message containing the possibly invalid value.

**Pull request merged**